includes/options/class-amp-user-options.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Class AMP_User_Options.
+ * 
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_User_Options
+ * 
+ * @since 1.6.0
+ */
+final class AMP_User_Options {
+
+	/**
+	 * Key for a user meta field storing preferences related to the plugin.
+	 *
+	 * @var string
+	 */
+	const USER_OPTIONS_KEY = 'amp_user_options';
+
+	/**
+	 * Key for the user option (stored in amp_features user meta) enabling or disabling developer tools.
+	 *
+	 * @var string
+	 */
+	const OPTION_DEVELOPER_TOOLS = 'developer_tools';
+
+	/**
+	 * Sets up hooks.
+	 */
+	public static function init() {
+		add_filter( 'amp_setup_wizard_data', [ __CLASS__, 'inject_setup_wizard_data' ] );
+		add_action( 'rest_api_init', [ __CLASS__, 'register_user_meta' ] );
+	}
+
+	/**
+	 * Registers user meta related to validation management.
+	 *
+	 * @since 1.6.0
+	 */
+	public static function register_user_meta() {
+		register_meta(
+			'user',
+			self::USER_OPTIONS_KEY,
+			[
+				'show_in_rest' => [
+					'schema' => [
+						'default'    => [
+							self::OPTION_DEVELOPER_TOOLS => 'unset',
+						],
+						'type'       => 'object',
+						'properties' => [
+							self::OPTION_DEVELOPER_TOOLS => [
+								'type' => 'string',
+								'enum' => [
+									'disabled',
+									'enabled',
+									'unset',
+								],
+							],
+						],
+					],
+				],
+				'single'       => true,
+				'type'         => 'object',
+			]
+		);
+	}
```

Comment:
I think we can just have a simple scalar user meta key for `amp_dev_tools_enabled`. It can be a boolean because it will either be on or off. If the user cannot `manage_options` (or per #2673 `amp_validate`) then it will always be `false`. It would be `true` by default for users who have that capability. In other words, I don't think we need to have a separate option that controls whether dev tools are enabled or disabled by default for users.